Index: iris/jabber/xmpp_ibb.cpp
===================================================================
--- iris/jabber/xmpp_ibb.cpp	(revision 534700)
+++ iris/jabber/xmpp_ibb.cpp	(working copy)
@@ -636,3 +636,4 @@
 	return d->mode;
 }
 
+#include "xmpp_ibb.moc"
Index: iris/jabber/filetransfer.cpp
===================================================================
--- iris/jabber/filetransfer.cpp	(revision 534700)
+++ iris/jabber/filetransfer.cpp	(working copy)
@@ -756,3 +756,5 @@
 	incoming(r);
 	return true;
 }
+
+#include "filetransfer.moc"
Index: iris/jabber/xmpp_jidlink.cpp
===================================================================
--- iris/jabber/xmpp_jidlink.cpp	(revision 534700)
+++ iris/jabber/xmpp_jidlink.cpp	(working copy)
@@ -318,3 +318,5 @@
 	if(j->setStream(bs))
 		d->incomingList.append(j);
 }
+
+#include "xmpp_jidlink.moc"
Index: iris/xmpp-im/xmpp_tasks.cpp
===================================================================
--- iris/xmpp-im/xmpp_tasks.cpp	(revision 534700)
+++ iris/xmpp-im/xmpp_tasks.cpp	(working copy)
@@ -2118,3 +2118,4 @@
 	return d->elem;
 }
 
+#include "xmpp_tasks.moc"
Index: iris/xmpp-core/tlshandler.cpp
===================================================================
--- iris/xmpp-core/tlshandler.cpp	(revision 534700)
+++ iris/xmpp-core/tlshandler.cpp	(working copy)
@@ -137,3 +137,5 @@
 	d->state = 0;
 	fail();
 }
+
+#include "tlshandler.moc"
Index: iris/xmpp-core/securestream.cpp
===================================================================
--- iris/xmpp-core/securestream.cpp	(revision 534700)
+++ iris/xmpp-core/securestream.cpp	(working copy)
@@ -306,8 +306,6 @@
 #endif
 };
 
-#include "securestream.moc"
-
 class SecureStream::Private
 {
 public:
@@ -589,3 +587,5 @@
 	if(bytesAvailable())
 		readyRead();
 }
+
+#include "securestream.moc"
Index: iris/xmpp-core/jid.cpp
===================================================================
--- iris/xmpp-core/jid.cpp	(revision 534700)
+++ iris/xmpp-core/jid.cpp	(working copy)
@@ -409,3 +409,5 @@
 	return true;*/
 	return StringPrepCache::resourceprep(s, 1024, norm);
 }
+
+#include "jid.moc"
Index: iris/xmpp-core/connector.cpp
===================================================================
--- iris/xmpp-core/connector.cpp	(revision 534700)
+++ iris/xmpp-core/connector.cpp	(working copy)
@@ -719,3 +719,5 @@
 {
 	httpSyncFinished();
 }
+
+#include "connector.moc"
Index: iris/xmpp-core/stream.cpp
===================================================================
--- iris/xmpp-core/stream.cpp	(revision 534700)
+++ iris/xmpp-core/stream.cpp	(working copy)
@@ -1765,3 +1765,5 @@
 		debug_ptr->incomingXml(e);
 }
 #endif
+
+#include "stream.moc"
Index: cutestuff/network/servsock.cpp
===================================================================
--- cutestuff/network/servsock.cpp	(revision 534700)
+++ cutestuff/network/servsock.cpp	(working copy)
@@ -108,3 +108,5 @@
 }
 
 // CS_NAMESPACE_END
+
+#include "servsock.moc"
Index: cutestuff/network/ndns.cpp
===================================================================
--- cutestuff/network/ndns.cpp	(revision 534700)
+++ cutestuff/network/ndns.cpp	(working copy)
@@ -377,3 +377,5 @@
 }
 
 // CS_NAMESPACE_END
+
+#include "ndns.moc"
Index: cutestuff/network/httppoll.cpp
===================================================================
--- cutestuff/network/httppoll.cpp	(revision 534700)
+++ cutestuff/network/httppoll.cpp	(working copy)
@@ -662,3 +662,5 @@
 }
 
 // CS_NAMESPACE_END
+
+#include "httppoll.moc"
Index: cutestuff/network/socks.cpp
===================================================================
--- cutestuff/network/socks.cpp	(revision 534700)
+++ cutestuff/network/socks.cpp	(working copy)
@@ -1221,3 +1221,5 @@
 }
 
 // CS_NAMESPACE_END
+
+#include "socks.moc"
Index: cutestuff/network/httpconnect.cpp
===================================================================
--- cutestuff/network/httpconnect.cpp	(revision 534700)
+++ cutestuff/network/httpconnect.cpp	(working copy)
@@ -366,3 +366,5 @@
 }
 
 // CS_NAMESPACE_END
+
+#include "httpconnect.moc"
Index: cutestuff/network/bsocket.cpp
===================================================================
--- cutestuff/network/bsocket.cpp	(revision 534700)
+++ cutestuff/network/bsocket.cpp	(working copy)
@@ -388,3 +388,5 @@
 }
 
 // CS_NAMESPACE_END
+
+#include "bsocket.moc"
Index: cutestuff/network/srvresolver.cpp
===================================================================
--- cutestuff/network/srvresolver.cpp	(revision 534700)
+++ cutestuff/network/srvresolver.cpp	(working copy)
@@ -318,3 +318,5 @@
 }
 
 // CS_NAMESPACE_END
+
+#include "srvresolver.moc"
Index: cutestuff/util/safedelete.cpp
===================================================================
--- cutestuff/util/safedelete.cpp	(revision 534700)
+++ cutestuff/util/safedelete.cpp	(working copy)
@@ -109,3 +109,5 @@
 {
 	delete this;
 }
+
+#include "safedelete.moc"
Index: cutestuff/util/bytestream.cpp
===================================================================
--- cutestuff/util/bytestream.cpp	(revision 534700)
+++ cutestuff/util/bytestream.cpp	(working copy)
@@ -267,3 +267,5 @@
 //! error is indicated by \a code.
 
 // CS_NAMESPACE_END
+
+#include "bytestream.moc"
