Index: iris/iris_export.h
===================================================================
--- iris/iris_export.h	(revision 0)
+++ iris/iris_export.h	(revision 0)
@@ -0,0 +1,8 @@
+#ifndef IRIS_EXPORT_H
+#define IRIS_EXPORT_H
+
+#include <QtGlobal>
+
+#define IRIS_EXPORT Q_DECL_EXPORT
+
+#endif
Index: cutestuff/cutestuff_export.h
===================================================================
--- cutestuff/cutestuff_export.h	(revision 0)
+++ cutestuff/cutestuff_export.h	(revision 0)
@@ -0,0 +1,8 @@
+#ifndef CUTESTUFF_EXPORT_H
+#define CUTESTUFF_EXPORT_H
+
+#include <QtGlobal>
+
+#define CUTESTUFF_EXPORT Q_DECL_EXPORT
+
+#endif

