Index: iris/jabber/filetransfer.cpp
===================================================================
--- iris/jabber/filetransfer.cpp	(revision 568436)
+++ iris/jabber/filetransfer.cpp	(working copy)
@@ -56,6 +56,7 @@
 	qlonglong size;
 	qlonglong sent;
 	QString desc;
+	QString preview;
 	bool rangeSupported;
 	qlonglong rangeOffset, rangeLength, length;
 	QString streamType;
@@ -104,13 +105,14 @@
 	d->proxy = proxy;
 }
 
-void FileTransfer::sendFile(const Jid &to, const QString &fname, qlonglong size, const QString &desc)
+void FileTransfer::sendFile(const Jid &to, const QString &fname, qlonglong size, const QString &desc, const QString& preview)
 {
 	d->state = Requesting;
 	d->peer = to;
 	d->fname = fname;
 	d->size = size;
 	d->desc = desc;
+	d->preview = preview;
 	d->sender = true;
 	d->id = d->m->link(this);
 
@@ -118,7 +120,7 @@
 	connect(d->ft, SIGNAL(finished()), SLOT(ft_finished()));
 	QStringList list;
 	list += "http://jabber.org/protocol/bytestreams";
-	d->ft->request(to, d->id, fname, size, desc, list);
+	d->ft->request(to, d->id, fname, size, desc, list,preview);
 	d->ft->go(true);
 }
 
@@ -171,6 +173,12 @@
 	return d->desc;
 }
 
+
+QString XMPP::FileTransfer::preview() const
+{
+	return d->preview;
+}
+
 bool FileTransfer::rangeSupported() const
 {
 	return d->rangeSupported;
@@ -312,6 +320,7 @@
 	d->fname = req.fname;
 	d->size = req.size;
 	d->desc = req.desc;
+	d->preview = req.preview;
 	d->rangeSupported = req.rangeSupported;
 }
 
@@ -461,7 +470,7 @@
 	delete d;
 }
 
-void JT_FT::request(const Jid &to, const QString &_id, const QString &fname, qlonglong size, const QString &desc, const QStringList &streamTypes)
+void JT_FT::request(const Jid &to, const QString &_id, const QString &fname, qlonglong size, const QString &desc, const QStringList &streamTypes, const QString& preview)
 {
 	QDomElement iq;
 	d->to = to;
@@ -480,6 +489,12 @@
 		de.appendChild(doc()->createTextNode(desc));
 		file.appendChild(de);
 	}
+	if(!preview.isEmpty()) {
+		QDomElement pr = doc()->createElement("preview");
+		pr.setAttribute("xmlns", "http://kopete.kde.org/protocol/file-preview");
+		pr.appendChild(doc()->createTextNode(preview));
+		file.appendChild(pr);
+	}
 	QDomElement range = doc()->createElement("range");
 	file.appendChild(range);
 	si.appendChild(file);
@@ -719,6 +734,11 @@
 	QDomElement de = file.elementsByTagName("desc").item(0).toElement();
 	if(!de.isNull())
 		desc = de.text();
+	
+	QString preview;
+	QDomElement pr = file.elementsByTagName("preview").item(0).toElement();
+	if(!pr.isNull())
+		preview= pr.text();
 
 	bool rangeSupported = false;
 	QDomElement range = file.elementsByTagName("range").item(0).toElement();
@@ -750,6 +770,7 @@
 	r.fname = fname;
 	r.size = size;
 	r.desc = desc;
+	r.preview = preview;
 	r.rangeSupported = rangeSupported;
 	r.streamTypes = streamTypes;
 
Index: iris/jabber/filetransfer.h
===================================================================
--- iris/jabber/filetransfer.h	(revision 568436)
+++ iris/jabber/filetransfer.h	(working copy)
@@ -39,7 +39,7 @@
 		void setProxy(const Jid &proxy);
 
 		// send
-		void sendFile(const Jid &to, const QString &fname, qlonglong size, const QString &desc);
+		void sendFile(const Jid &to, const QString &fname, qlonglong size, const QString &desc, const QString& preview=QString());
 		qlonglong offset() const;
 		qlonglong length() const;
 		int dataSizeNeeded() const;
@@ -50,6 +50,7 @@
 		QString fileName() const;
 		qlonglong fileSize() const;
 		QString description() const;
+		QString preview() const;
 		bool rangeSupported() const;
 		void accept(qlonglong offset=0, qlonglong length=0);
 
@@ -123,7 +124,7 @@
 		JT_FT(Task *parent);
 		~JT_FT();
 
-		void request(const Jid &to, const QString &id, const QString &fname, qlonglong size, const QString &desc, const QStringList &streamTypes);
+		void request(const Jid &to, const QString &id, const QString &fname, qlonglong size, const QString &desc, const QStringList &streamTypes, const QString &preview=QString());
 		qlonglong rangeOffset() const;
 		qlonglong rangeLength() const;
 		QString streamType() const;
@@ -143,6 +144,7 @@
 		QString fname;
 		qlonglong size;
 		QString desc;
+		QString preview;
 		bool rangeSupported;
 		QStringList streamTypes;
 	};
