Index: iris/include/xmpp.h
===================================================================
--- iris/include/xmpp.h	(Revision 636263)
+++ iris/include/xmpp.h	(Revision 636334)
@@ -37,7 +37,7 @@
 namespace QCA
 {
 	class TLS;
-};
+}
 
 #ifndef CS_XMPP
 class ByteStream;
@@ -524,6 +524,6 @@
 		void handleError();
 		void srvProcessNext();
 	};
-};
+}
 
 #endif
Index: iris/include/xmpp_xdata.h
===================================================================
--- iris/include/xmpp_xdata.h	(Revision 636263)
+++ iris/include/xmpp_xdata.h	(Revision 636334)
@@ -147,6 +147,6 @@
 		QSharedDataPointer<Private> d;
 	};
 
-};
+}
 
 #endif
Index: iris/xmpp-im/xmpp_xmlcommon.h
===================================================================
--- iris/xmpp-im/xmpp_xmlcommon.h	(Revision 636263)
+++ iris/xmpp-im/xmpp_xmlcommon.h	(Revision 636334)
@@ -67,6 +67,6 @@
 	void readBoolAttribute(QDomElement e, const QString &name, bool *v);
 
 	//QString tagContent(const QDomElement &e); // obsolete;
-};
+}
 
 #endif
Index: iris/xmpp-im/xmpp_xmlcommon.cpp
===================================================================
--- iris/xmpp-im/xmpp_xmlcommon.cpp	(Revision 636263)
+++ iris/xmpp-im/xmpp_xmlcommon.cpp	(Revision 636334)
@@ -468,5 +468,5 @@
 	}
 }
 
-};
+}
 
Index: iris/xmpp-core/compress.cpp
===================================================================
--- iris/xmpp-core/compress.cpp	(Revision 636263)
+++ iris/xmpp-core/compress.cpp	(Revision 636334)
@@ -1,6 +1,7 @@
 #include <QObject>
 #include <QIODevice>
 #include <zlib.h>
+#include <stdlib.h>
 
 #include "compress.h"
 
Index: iris/xmpp-core/parser.cpp
===================================================================
--- iris/xmpp-core/parser.cpp	(Revision 636263)
+++ iris/xmpp-core/parser.cpp	(Revision 636334)
@@ -547,7 +547,7 @@
 		Q3PtrList<Parser::Event> eventList;
 		bool needMore;
 	};
-};
+}
 
 
 //----------------------------------------------------------------------------
