Index: iris/xmpp-im/xmpp_xmlcommon.h
===================================================================
--- iris/xmpp-im/xmpp_xmlcommon.h	(revision 772486)
+++ iris/xmpp-im/xmpp_xmlcommon.h	(working copy)
@@ -38,7 +38,7 @@
 QDomElement IRIS_EXPORT findSubTag(const QDomElement &e, const QString &name, bool *found);
 QDomElement IRIS_EXPORT createIQ(QDomDocument *doc, const QString &type, const QString &to, const QString &id);
 QDomElement IRIS_EXPORT queryTag(const QDomElement &e);
-QString queryNS(const QDomElement &e);
+QString IRIS_EXPORT queryNS(const QDomElement &e);
 void getErrorFromElement(const QDomElement &e, const QString &baseNS, int *code, QString *str);
 QDomElement addCorrectNS(const QDomElement &e);
 
