Index: iris/xmpp/zlib/zlibdecompressor.cpp
===================================================================
--- iris/xmpp/zlib/zlibdecompressor.cpp	(revision 928428)
+++ iris/xmpp/zlib/zlibdecompressor.cpp	(working copy)
@@ -33,7 +33,7 @@
 	write(QByteArray(),true);
 	int result = inflateEnd(zlib_stream_);
 	if (result != Z_OK) 
-		qWarning(QString("compressor.c: inflateEnd failed (%1)").arg(result).toAscii());
+		qWarning() << QString("compressor.c: inflateEnd failed (%1)").arg(result).toAscii();
 	
 	flushed_ = true;
 }
@@ -58,7 +58,7 @@
 		zlib_stream_->next_out = (Bytef*) (output.data() + output_position);
 		result = inflate(zlib_stream_,(flush ? Z_FINISH : Z_NO_FLUSH));
 		if (result == Z_STREAM_ERROR) {
-			qWarning(QString("compressor.cpp: Error ('%1')").arg(zlib_stream_->msg).toAscii());
+			qWarning() << QString("compressor.cpp: Error ('%1')").arg(zlib_stream_->msg).toAscii();
 			return result;
 		}
 		output_position += CHUNK_SIZE;
@@ -79,7 +79,7 @@
 			zlib_stream_->next_out = (Bytef*) (output.data() + output_position);
 			result = inflate(zlib_stream_,Z_SYNC_FLUSH);
 			if (result == Z_STREAM_ERROR) {
-				qWarning(QString("compressor.cpp: Error ('%1')").arg(zlib_stream_->msg).toAscii());
+				qWarning() << QString("compressor.cpp: Error ('%1')").arg(zlib_stream_->msg).toAscii();
 				return result;
 			}
 			output_position += CHUNK_SIZE;
Index: iris/xmpp/zlib/zlibcompressor.cpp
===================================================================
--- iris/xmpp/zlib/zlibcompressor.cpp	(revision 928428)
+++ iris/xmpp/zlib/zlibcompressor.cpp	(working copy)
@@ -33,7 +33,7 @@
 	write(QByteArray(),true);
 	int result = deflateEnd(zlib_stream_);
 	if (result != Z_OK) 
-		qWarning(QString("compressor.c: deflateEnd failed (%1)").arg(result).toAscii());
+		qWarning() << QString("compressor.c: deflateEnd failed (%1)").arg(result).toAscii();
 	
 	flushed_ = true;
 }
@@ -58,7 +58,7 @@
 		zlib_stream_->next_out = (Bytef*) (output.data() + output_position);
 		result = deflate(zlib_stream_,(flush ? Z_FINISH : Z_NO_FLUSH));
 		if (result == Z_STREAM_ERROR) {
-			qWarning(QString("compressor.cpp: Error ('%1')").arg(zlib_stream_->msg).toAscii());
+			qWarning() << QString("compressor.cpp: Error ('%1')").arg(zlib_stream_->msg).toAscii();
 			return result;
 		}
 		output_position += CHUNK_SIZE;
@@ -77,7 +77,7 @@
 			zlib_stream_->next_out = (Bytef*) (output.data() + output_position);
 			result = deflate(zlib_stream_,Z_SYNC_FLUSH);
 			if (result == Z_STREAM_ERROR) {
-				qWarning(QString("compressor.cpp: Error ('%1')").arg(zlib_stream_->msg).toAscii());
+				qWarning() << QString("compressor.cpp: Error ('%1')").arg(zlib_stream_->msg).toAscii();
 				return result;
 			}
 			output_position += CHUNK_SIZE;
