Index: dialog.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/libkopete/compat/ksettings/dialog.cpp,v
retrieving revision 1.1
diff -u -p -r1.1 dialog.cpp
--- dialog.cpp	3 Sep 2003 08:50:34 -0000	1.1
+++ dialog.cpp	7 Sep 2003 11:05:56 -0000
@@ -33,6 +33,8 @@
 #include "ksettings/dispatcher.h"
 #include "ksettings/componentsdialog.h"
 
+#include <kdeversion.h>
+
 namespace KSettings
 {
 
@@ -591,9 +591,11 @@ void Dialog::createDialogFromServices()
 	// I have no idea how to check that in KPluginSelector::load()...
 	//d->dlg->showButton( KDialogBase::User1, true );
 
+#if KDE_IS_VERSION( 3, 1, 90 )
 	if( ! d->staticlistview )
 		d->dlg->addButtonBelowList( i18n( "Configure..." ), this,
 			SLOT( configureTree() ) );
+#endif
 
 	connect( d->dlg, SIGNAL( okClicked() ), Dispatcher::self(),
 		SLOT( syncConfiguration() ) );
@@ -601,8 +601,10 @@ void Dialog::createDialogFromServices()
 
 	d->pagetree.addToDialog( d->dlg );
 
+#if KDE_IS_VERSION( 3, 1, 90 )
 	if( dialogface == KJanusWidget::TreeList )
 		d->dlg->unfoldTreeList( true );
+#endif
 }
 
 void Dialog::configureTree()
