Index: dialog.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/libkopete/compat/ksettings/dialog.cpp,v
retrieving revision 1.4
diff -u -p -b -B -r1.4 dialog.cpp
--- dialog.cpp	9 Sep 2003 18:13:23 -0000	1.4
+++ dialog.cpp	9 Sep 2003 20:14:56 -0000
@@ -552,8 +552,7 @@ void Dialog::createDialogFromServices()
 		// we create the KCModuleInfo
 		KCModuleInfo * info = new KCModuleInfo( *it );
 		QString parentid;
-		QVariant tmp = info->service()->property( "X-KDE-CfgDlgHierarchy",
-			QVariant::String );
+		QVariant tmp = info->service()->property( "X-KDE-CfgDlgHierarchy" );
 		if( tmp.isValid() )
 			parentid = tmp.toString();
 		d->pagetree.insert( info, parentid );
