--- index-kopete-svn.docbook	2008-02-25 17:24:01.000000000 +0100
+++ index-kopete-svn-with-otr.docbook	2008-02-25 17:27:39.000000000 +0100
@@ -828,7 +828,167 @@
 </varlistentry>
 </variablelist>
 </sect2>
-
+<sect2 id="plugins-otr">
+<title>OTR</title>
+<para>The <guilabel>OTR</guilabel> plugin allows you to encrypt conversations using the Off-The-Record encryption method. This encryption method basically works like the PGP encryption but it is optimized for instant messaging and handles the key excange on its own. The only thing you have to do, is to make sure that you really are writing to the person you expect by using one of OTR's authentication mechanisms.</para>
+<sect3 id="plugins-otr-menus">
+<title>Menu items</title>
+<para>The following item is added to the Contact List's menus:</para>
+<variablelist>
+<varlistentry>
+<term>
+<menuchoice>
+<guimenu>Edit</guimenu>
+<guimenuitem>OTR Policy</guimenuitem>
+</menuchoice>
+</term>
+<listitem>
+<para>(Enabled when a contact is selected) You can set the encryption policy for each contact here. Setting this entry to "Default" causes Kopete to use the default setting in the configure dialog</para>
+</listitem>
+</varlistentry>
+</variablelist>
+<para>The following items are added to the Chat window's menus:</para>
+<variablelist>
+<varlistentry>
+<term>
+<menuchoice>
+<guimenu>Tools</guimenu>
+<guimenuitem>OTR Settings</guimenuitem>
+<guimenuitem>Start OTR session</guimenuitem>
+</menuchoice>
+</term>
+<listitem>
+<para>This lets you initiate an OTR encrypted session.</para>
+</listitem>
+</varlistentry>
+<varlistentry>
+<term>
+<menuchoice>
+<guimenu>Tools</guimenu>
+<guimenuitem>OTR Settings</guimenuitem>
+<guimenuitem>End OTR session</guimenuitem>
+</menuchoice>
+</term>
+<listitem>
+<para>This lets you end a previously established OTR session.</para>
+</listitem>
+</varlistentry>
+<varlistentry>
+<term>
+<menuchoice>
+<guimenu>Tools</guimenu>
+<guimenuitem>OTR Settings</guimenuitem>
+<guimenuitem>Authenticate contact</guimenuitem>
+</menuchoice>
+</term>
+<listitem>
+<para>Use this menu entry if you wish to authenticate the contact.</para>
+</listitem>
+</varlistentry>
+</variablelist>
+</sect3>
+<sect3 id="plugins-otr-setup">
+<title>The settings dialog</title>
+<para>In the setting dialog you have various options to set the modules behaviour during your chatsessions. The most important of them ist the default policy configuration. Set the policy to "Opportunistic" if you wish Kopete to check if you contacts support OTR and establish an encrypted session on its own or to "Manual" if you wish to control each encryption state on your own. The options "Always" and "Never" are recommended only for experienced users that wish to set the policy on a per-contact base. In the "Private Keys" Pane you have the possibility to create you personal keypair for each account. You don't need to do this manually. If an account does not have a keypair it will be generated automatically the first time it is required. In the "Known Fingerprints" tab you have the possibility to see, verify and delete known public fingerprints of your contacts.</para>
+</sect3>
+<sect3 id="plugins-otr-auth">
+<title>Authentication</title>
+<para>You've probably received email from people pretending to be banks, credit agencies, even wealthy Nigerian expatriates. People lie about who they are all the time on the Internet. Authentication is a way to make sure that nobody can lie to you about who they are when they use OTR.</para>
+<sect4 id="plugins-otr-auth-when">
+<title>When to authenticate</title>
+<para>You should authenticate a buddy the very first time that you talk to them using OTR. If you don't, then you can't really be sure that someone else isn't impersonating them or trying to listen in on your conversation. However, once you've authenticated your buddy once, you don't have to do it again. OTR will automatically do the authentication for all of your future conversations with that buddy. </para>
+<para>The only exceptions occur when your buddy switches between multiple computers or multiple IM accounts. In this case, you will need to authenticate once for each computer and account. Once you've done this, your buddy can freely use any of the computers you've authenticated them on, and OTR will recognize them automatically. If your buddy uses a new computer or account that OTR does not recognize, a message will pop up in your conversation window telling you about it.</para>
+<screenshot>
+  <mediaobject>
+    <imageobject>
+      <imagedata fileref="otr-auth-1.png" format="PNG"/>
+    </imageobject>
+    <textobject>
+      <phrase>When to authenticate</phrase>
+    </textobject>
+  </mediaobject>
+</screenshot>
+</sect4>
+<sect4 id="plugins-otr-auth-how">
+<title>How to authenticate</title>
+<para>Important:If you are using OTR version 3.0.0 or earlier, or you would like to continue using the old fingerprints method, please refer to the help on fingerprints. The remainder of this page assumes that you are using a recent copy of OTR, and the standard authentication method. </para>
+<para>To authenticate someone, open a conversation with them and click on "Authenticate Contact" on the OTR button. OTR will ask both you and your friend to enter a secret known only to the two of you.</para>
+<screenshot>
+  <mediaobject>
+    <imageobject>
+      <imagedata fileref="otr-auth-2.png" format="PNG"/>
+    </imageobject>
+    <textobject>
+      <phrase>How to authenticate</phrase>
+    </textobject>
+  </mediaobject>
+</screenshot>
+<para>If you both enter the same thing, then you know that you are really talking to your friend. Any imposter should have a hard time guessing what you're typing in, so you'll be able to catch them in the act. </para>
+<para>This method of authentication is only effective if it's hard for an imposter to guess what you're typing, but easy for your friend.
+The following is an example of what not to do:</para>
+<screenshot>
+  <mediaobject>
+    <imageobject>
+      <imagedata fileref="otr-auth-3.png" format="PNG"/>
+    </imageobject>
+    <textobject>
+      <phrase>Bad Example</phrase>
+    </textobject>
+  </mediaobject>
+</screenshot>
+<para>Here you are telling the other person exactly what to do. An imposter can figure out what to type in just as easily as your friend can.
+A better way to pick a secret would be something like this:</para>
+ <screenshot>
+  <mediaobject>
+    <imageobject>
+      <imagedata fileref="otr-auth-4.png" format="PNG"/>
+    </imageobject>
+    <textobject>
+      <phrase>Good Example</phrase>
+    </textobject>
+  </mediaobject>
+</screenshot>
+<para>Now an imposter has no idea what to type in. Your friend, on the other hand, knows exactly what you're talking about. When OTR tells you that the secrets matched, you can be sure that you are really talking to your friend. </para>
+</sect4>
+<sect4 id="plugins-otr-auth-results">
+<title>What the results mean</title>
+<para>When you have entered your secret and hit OK, the message "Authenticatig contact..." appears in the chatwindow. After a few seconds the result should show up in the chatwindow and display one of the following messages:</para>
+<screenshot>
+  <mediaobject>
+    <imageobject>
+      <imagedata fileref="otr-auth-5.png" format="PNG"/>
+    </imageobject>
+    <textobject>
+      <phrase>Authentication successful</phrase>
+    </textobject>
+  </mediaobject>
+</screenshot>
+<para>This means that authentication has been a complete success. Your buddy entered the same secret that you did, and so they are not an imposter. The OTR button will automatically change to "Private", showing that conversations with this buddy are safe.</para>
+<screenshot>
+  <mediaobject>
+    <imageobject>
+      <imagedata fileref="otr-auth-6.png" format="PNG"/>
+    </imageobject>
+    <textobject>
+      <phrase>Authentication failed</phrase>
+    </textobject>
+  </mediaobject>
+</screenshot>
+<para>This means that although there were no errors, your buddy did not enter the same text as you. You should try again, making sure that you are clear about what to type (for example, "the restaurant name in lower case"). If you repeatedly get this result, you should view your buddy with suspicion.</para>
+<screenshot>
+  <mediaobject>
+    <imageobject>
+      <imagedata fileref="otr-auth-7.png" format="PNG"/>
+    </imageobject>
+    <textobject>
+      <phrase>Authentication error</phrase>
+    </textobject>
+  </mediaobject>
+</screenshot>
+<para>This means that something has gone wrong and the process could not complete normally. This will happen if your buddy hits "cancel" or fails to receive one of your messages. In this case, you should simply try again. If you try several times and keep getting an error, you should view your buddy with suspicion.</para>
+</sect4>
+</sect3>
+</sect2>
 <sect2 id="plugins-kopetex">
 <title>KopeteTeX</title>
 <para>KopeteTeX allows scientists and mathematicians to hold conversations using the LaTeX markup language.  Expressions entered within $$ are rendered as a graphic in the chatwindow, and can be cut and pasted as the original Latex.  To use this plugin you must have LaTeX installed</para>
