? less-warning1.patch
Index: libyahoo2.c
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/yahoo/libyahoo2/libyahoo2.c,v
retrieving revision 1.15
diff -u -p -r1.15 libyahoo2.c
--- libyahoo2.c	29 Sep 2003 06:24:07 -0000	1.15
+++ libyahoo2.c	23 Oct 2003 01:34:23 -0000
@@ -101,7 +101,7 @@ void yahoo_register_callbacks(struct yah
 #define YAHOO_CALLBACK(x)	x
 #endif
 
-int yahoo_log_message(char * fmt, ...)
+int yahoo_log_message(const char * fmt, ...)
 {
 	char out[1024];
 	va_list ap;
Index: yahoo2_callbacks.h
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/yahoo/libyahoo2/yahoo2_callbacks.h,v
retrieving revision 1.7
diff -u -p -r1.7 yahoo2_callbacks.h
--- yahoo2_callbacks.h	29 Sep 2003 06:24:07 -0000	1.7
+++ yahoo2_callbacks.h	23 Oct 2003 01:34:23 -0000
@@ -534,7 +534,7 @@ void YAHOO_CALLBACK_TYPE(ext_yahoo_webca
  * Returns:
  * 	0
  */
-int YAHOO_CALLBACK_TYPE(ext_yahoo_log)(char *fmt, ...);
+int YAHOO_CALLBACK_TYPE(ext_yahoo_log)(const char *fmt, ...);
 
 
 
Index: yahoo_debug.h
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/yahoo/libyahoo2/yahoo_debug.h,v
retrieving revision 1.4
diff -u -p -r1.4 yahoo_debug.h
--- yahoo_debug.h	12 Sep 2003 06:39:54 -0000	1.4
+++ yahoo_debug.h	23 Oct 2003 01:34:23 -0000
@@ -19,7 +19,7 @@
  *
  */
 
-extern int yahoo_log_message(char *fmt, ...);
+extern int yahoo_log_message(const char *fmt, ...);
 
 #define NOTICE(x) if(log_level >= YAHOO_LOG_NOTICE) { yahoo_log_message x; yahoo_log_message("\n"); }
 
Index: yahoo_util.c
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/yahoo/libyahoo2/yahoo_util.c,v
retrieving revision 1.4
diff -u -p -r1.4 yahoo_util.c
--- yahoo_util.c	12 Sep 2003 06:39:54 -0000	1.4
+++ yahoo_util.c	23 Oct 2003 01:34:23 -0000
@@ -39,7 +39,7 @@ char *strchr (), *strrchr ();
 
 #include "yahoo_util.h"
 
-char * y_string_append(char * string, char * append)
+char * y_string_append(char * string, const char * append)
 {
 	int size = strlen(string) + strlen(append) + 1;
 	char * new_string = y_renew(char, string, size);
Index: yahoo_util.h
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/yahoo/libyahoo2/yahoo_util.h,v
retrieving revision 1.8
diff -u -p -r1.8 yahoo_util.h
--- yahoo_util.h	29 Sep 2003 23:27:31 -0000	1.8
+++ yahoo_util.h	23 Oct 2003 01:34:23 -0000
@@ -86,7 +86,7 @@ void y_strfreev(char ** vector);
  * The following three functions return newly allocated memory.
  * You must free it yourself
  */
-char * y_string_append(char * str, char * append);
+char * y_string_append(char * str, const char * append);
 char * y_str_to_utf8(const char * in);
 char * y_utf8_to_str(const char * in);
 
