Index: kpluginselector.cpp
===================================================================
RCS file: /home/kde/kdenonbeta/kopete/libkopete/compat/kcdialog/kpluginselector.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -p -r1.1 -r1.2
--- kpluginselector.cpp	27 Aug 2003 17:35:59 -0000	1.1
+++ kpluginselector.cpp	27 Aug 2003 19:08:26 -0000	1.2
@@ -82,7 +82,7 @@
 #include <qvaluelist.h>
 #include <kservice.h>
 #include <ktrader.h>
-#include <ktabwidget.h>
+#include "ktabwidget.h"
 #include <kiconloader.h>
 #include <kcmodule.h>
 #include "kcmoduleinfo.h"
--- /home/martijn/src/kde-head/kdelibs/kutils/kcmultidialog.cpp	2003-09-09 15:25:25.000000000 +0200
+++ kcmultidialog.cpp	2003-09-09 20:08:15.000000000 +0200
@@ -103,7 +103,7 @@
                 m->setChanged( false );
             }
             QStringList * names = moduleParentComponents[ m ];
-            kdDebug(710) << k_funcinfo << *names << " saved and added to the list" << endl;
+            //kdDebug(710) << k_funcinfo << *names << " saved and added to the list" << endl;
             for( QStringList::ConstIterator it = names->begin(); it != names->end(); ++it )
                 if( updatedModules.find( *it ) == updatedModules.end() )
                     updatedModules.append( *it );
@@ -157,6 +157,7 @@
     enableButton( Apply, false );
 }
 
+#if 0 // I don't think Kopete needs this - Martijn
 void KCMultiDialog::addModule(const QString& path, bool withfallback)
 {
     kdDebug(710) << "KCMultiDialog::addModule " << path << endl;
@@ -170,6 +171,7 @@
     KCModuleInfo info(s);
     addModule(info, QStringList(), withfallback);
 }
+#endif
 
 void KCMultiDialog::addModule(const KCModuleInfo& moduleinfo,
         QStringList parentmodulenames, bool withfallback)
--- /home/martijn/src/kde-head/kdelibs/kutils/kcmoduleinfo.cpp	2003-09-09 15:25:25.000000000 +0200
+++ kcmoduleinfo.cpp	2003-09-09 20:08:30.000000000 +0200
@@ -29,6 +29,7 @@
 #include <kglobal.h>
 #include <kstandarddirs.h>
 
+#if 0 // I don't think Kopete needs this - Martijn
 KCModuleInfo::KCModuleInfo(const QString& desktopFile)
   : _fileName(desktopFile), d(0L)
 {
@@ -37,6 +38,7 @@
   //kdDebug(1208) << "desktopFile = " << desktopFile << endl;
   init( KService::serviceByStorageId(desktopFile) );
 }
+#endif
 
 KCModuleInfo::KCModuleInfo( KService::Ptr moduleInfo )
   : _fileName( moduleInfo->desktopEntryPath() )
