Index: kcmoduleinfo.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/libkopete/compat/kcdialog/kcmoduleinfo.cpp,v
retrieving revision 1.6
diff -u -p -b -B -r1.6 kcmoduleinfo.cpp
--- kcmoduleinfo.cpp	9 Sep 2003 18:13:23 -0000	1.6
+++ kcmoduleinfo.cpp	9 Sep 2003 20:14:32 -0000
@@ -111,25 +111,25 @@ KCModuleInfo::loadAll() 
   _allLoaded = true;
 
   // library and factory
-  setHandle(_service->property("X-KDE-FactoryName", QVariant::String).toString());
+  setHandle(_service->property("X-KDE-FactoryName").toString());
 
   QVariant tmp;
 
   // read weight
-  tmp = _service->property( "X-KDE-Weight", QVariant::Int );
+  tmp = _service->property( "X-KDE-Weight" );
   setWeight( tmp.isValid() ? tmp.toInt() : -1 );
 
   // does the module need super user privileges?
-  tmp = _service->property( "X-KDE-RootOnly", QVariant::Bool );
+  tmp = _service->property( "X-KDE-RootOnly" );
   setNeedsRootPrivileges( tmp.isValid() ? tmp.toBool() : false );
 
   // does the module need to be shown to root only?
   // Deprecated !
-  tmp = _service->property( "X-KDE-IsHiddenByDefault", QVariant::Bool );
+  tmp = _service->property( "X-KDE-IsHiddenByDefault");
   setIsHiddenByDefault( tmp.isValid() ? tmp.toBool() : false );
 
   // get the documentation path
-  setDocPath( _service->property( "DocPath", QVariant::String ).toString() );
+  setDocPath( _service->property( "DocPath" ).toString() );
 }
 
 QString
Index: kcmultidialog.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/libkopete/compat/kcdialog/kcmultidialog.cpp,v
retrieving revision 1.8
diff -u -p -b -B -r1.8 kcmultidialog.cpp
--- kcmultidialog.cpp	9 Sep 2003 18:13:23 -0000	1.8
+++ kcmultidialog.cpp	9 Sep 2003 20:14:32 -0000
@@ -243,8 +243,7 @@ void KCMultiDialog::slotAboutToShow(QWid
 
     QStringList parentComponents = loadInfo->info.service()->property(
             "X-KDE-ParentComponents" ).toStringList();
-    kdDebug(710) << k_funcinfo << "ParentComponents=" << parentComponents
-        << endl;
+//    kdDebug(710) << k_funcinfo << "ParentComponents=" << parentComponents << endl;
     moduleParentComponents.insert( module,
             new QStringList( parentComponents ) );
     module->reparent(page,0,QPoint(0,0),true);
Index: ktabbar.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/libkopete/compat/kcdialog/ktabbar.cpp,v
retrieving revision 1.2
diff -u -p -b -B -r1.2 ktabbar.cpp
--- ktabbar.cpp	9 Sep 2003 18:13:23 -0000	1.2
+++ ktabbar.cpp	9 Sep 2003 20:14:32 -0000
@@ -326,9 +326,14 @@ void KTabBar::paintLabel( QPainter *p, c
         int pixh = pixmap.height();
         r.setLeft( r.left() + pixw + 4 );
         r.setRight( r.right() + 2 );
-
+#if QT_VERSION >= 0x030200
         int inactiveXShift = style().pixelMetric( QStyle::PM_TabBarTabShiftHorizontal, this );
         int inactiveYShift = style().pixelMetric( QStyle::PM_TabBarTabShiftVertical, this );
+#else  //FIXME! WARNING  i don't know if i can set this to 0
+        int inactiveXShift = 0;
+        int inactiveYShift = 0;
+#endif
+
 
         p->drawPixmap( br.left() + 2 + ((selected == TRUE) ? 0 : inactiveXShift),
                          br.center().y()-pixh/2 + ((selected == TRUE) ? 0 : inactiveYShift),
