Index: kpluginselector.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/libkopete/compat/kcdialog/kpluginselector.cpp,v
retrieving revision 1.7
retrieving revision 1.9
diff -u -p -r1.7 -r1.9
--- kpluginselector.cpp	16 Sep 2003 18:01:51 -0000	1.7
+++ kpluginselector.cpp	16 Oct 2003 15:28:29 -0000	1.9
@@ -33,6 +33,7 @@
 #include <qobjectlist.h>
 #include <qcstring.h>
 
+#include <kdeversion.h>
 #include <kdebug.h>
 #include <klocale.h>
 #include <klistview.h>
