diff -u /home/till/kde/libxmpp/xmpp/src/xmpp_stream.cpp ./xmpp_stream.cpp
--- /home/till/kde/libxmpp/xmpp/src/xmpp_stream.cpp	2003-05-26 03:22:58.000000000 +0200
+++ ./xmpp_stream.cpp	2003-08-26 02:55:49.000000000 +0200
@@ -18,7 +18,7 @@
  *
  */
 
-//#define QSSL_STATIC
+#define QSSL_STATIC
 
 #include"xmpp_stream.h"
 
@@ -32,6 +32,7 @@
 #include"ndns.h"
 #include"xmpp_xmlfilter.h"
 #include"qssl.h"
+#include"qssl_p.h"
 
 using namespace Jabber;
 
@@ -208,7 +209,7 @@
 }
 
 //! \brief Returns if this is a warning or error
-//! 
+//!
 //! \return true: if warning\n
 //!   false: if error
 bool StreamError::isWarning() const
@@ -326,7 +327,7 @@
 	ssl_tried = true;
 
 #if defined(QSSL_STATIC)
-	qssl = new QSSL;
+	qssl = new _QSSL;
 #else
 	if(lib)
 		return true;
@@ -440,8 +441,11 @@
 
 bool Stream::loadSSL(const QStringList &dirs)
 {
+
+#if !defined(QSSL_STATIC)
 	if(qssl)
 		QSSL_unload();
+#endif
 
 	sslErrorLog = "";
 	for(QStringList::ConstIterator it = dirs.begin(); it != dirs.end(); ++it) {
