Index: ndns.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/cutestuff/network/ndns.cpp,v
retrieving revision 1.2
retrieving revision 1.3
diff -u -r1.2 -r1.3
--- ndns.cpp	29 Aug 2003 11:18:07 -0000	1.2
+++ ndns.cpp	30 Aug 2003 19:39:37 -0000	1.3
@@ -258,3 +258,4 @@
 
 	QApplication::postEvent(par, new NDnsWorkerEvent(this));
 }
+#include "ndns.moc"
Index: qssl.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/cutestuff/ssl/qssl.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- qssl.cpp	26 Aug 2003 00:55:48 -0000	1.1
+++ qssl.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -912,3 +912,5 @@
 	}
 	return rc;
 }
+
+#include "qssl.moc"
Index: xmpp_bytestream.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_bytestream.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- xmpp_bytestream.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ xmpp_bytestream.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -68,3 +68,5 @@
 {
 	return 0;
 }
+
+#include "xmpp_bytestream.moc"
Index: xmpp_client.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_client.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- xmpp_client.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ xmpp_client.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -1310,3 +1310,5 @@
 	}
 	return it;
 }
+
+#include "xmpp_client.moc"
Index: xmpp_dtcp.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_dtcp.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- xmpp_dtcp.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ xmpp_dtcp.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -1651,3 +1651,5 @@
 {
 	connectionReady(x);
 }
+
+#include "xmpp_dtcp.moc"
Index: xmpp_ibb.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_ibb.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- xmpp_ibb.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ xmpp_ibb.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -634,3 +634,5 @@
 {
 	return d->mode;
 }
+
+#include "xmpp_ibb.moc"
Index: xmpp_jidlink.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_jidlink.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- xmpp_jidlink.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ xmpp_jidlink.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -342,3 +342,5 @@
 	// TODO handle shutdown of this object
 	bs->deleteLater();
 }
+
+#include "xmpp_jidlink.moc"
Index: xmpp_stream.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_stream.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- xmpp_stream.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ xmpp_stream.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -1089,3 +1089,5 @@
 	e.save(ts, 0);
 	return out.utf8();
 }
+
+#include "xmpp_stream.moc"
Index: xmpp_tasks.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_tasks.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- xmpp_tasks.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ xmpp_tasks.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -1721,3 +1721,5 @@
 	return true;
 }
 
+
+#include "xmpp_tasks.moc"
Index: xmpp_xmlfilter.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_xmlfilter.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- xmpp_xmlfilter.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ xmpp_xmlfilter.cpp	30 Aug 2003 19:39:38 -0000	1.2
@@ -232,3 +232,5 @@
 
 	return true;
 }
+
+#include "xmpp_xmlfilter.moc"
