Index: libxmpp/xmpp/src/xmpp_xmlcommon.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_xmlcommon.cpp,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- libxmpp/xmpp/src/xmpp_xmlcommon.cpp	26 Aug 2003 00:55:49 -0000	1.1
+++ libxmpp/xmpp/src/xmpp_xmlcommon.cpp	13 Sep 2003 10:44:57 -0000	1.2
@@ -382,5 +382,5 @@
 	}
 }
 
-};
+}
 
Index: libxmpp/xmpp/src/xmpp_xmlcommon.h
===================================================================
RCS file: /home/kde/kdenetwork/kopete/protocols/jabber/libxmpp/xmpp/src/xmpp_xmlcommon.h,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -r1.1 -r1.2
--- libxmpp/xmpp/src/xmpp_xmlcommon.h	26 Aug 2003 00:55:49 -0000	1.1
+++ libxmpp/xmpp/src/xmpp_xmlcommon.h	13 Sep 2003 10:44:57 -0000	1.2
@@ -68,6 +68,6 @@
 	void readBoolAttribute(QDomElement e, const QString &name, bool *v);
 
 	//QString tagContent(const QDomElement &e); // obsolete;
-};
+}
 
 #endif
