Index: kautoconfig.cpp
===================================================================
RCS file: /home/kde/kdenetwork/kopete/libkopete/compat/kautoconfig.cpp,v
retrieving revision 1.3
diff -u -p -b -B -r1.3 kautoconfig.cpp
--- kautoconfig.cpp	9 Sep 2003 18:13:22 -0000	1.3
+++ kautoconfig.cpp	9 Sep 2003 20:13:45 -0000
@@ -27,6 +27,7 @@
 #endif
 #include <kconfig.h>
 #include <kapplication.h>
+#include <kdeversion.h>
 
 class KAutoConfig::KAutoConfigPrivate {
 
@@ -195,19 +196,22 @@ bool KAutoConfig::saveSettings() {
       ++it;
       QVariant defaultValue = d->defaultValues[groupWidget];
       QVariant currentValue = propertyMap->property(groupWidget);
-
+#if KDE_IS_VERSION( 3, 1, 90 )
       if(!config->hasDefault(groupWidget->name()) && currentValue == defaultValue){
         config->revertToDefault(groupWidget->name());
         widgetChanged = true;
       }
       else{
+#endif
         QVariant savedValue = config->readPropertyEntry(groupWidget->name(),
                                                              defaultValue);
         if(savedValue != currentValue){
           config->writeEntry(groupWidget->name(), currentValue);
           widgetChanged = true;
         }
+#if KDE_IS_VERSION( 3, 1, 90 )
       }
+#endif
     }
     d->changed |= widgetChanged;
     if(widgetChanged)
