diff --git a/src/xmpp/xmpp-im/xmpp_vcard.cpp b/src/xmpp/xmpp-im/xmpp_vcard.cpp
index 18aa8b4..3ae2475 100644
--- a/src/xmpp/xmpp-im/xmpp_vcard.cpp
+++ b/src/xmpp/xmpp-im/xmpp_vcard.cpp
@@ -549,7 +549,7 @@ bool VCard::fromXml(const QDomElement &q)
 		else if ( tag == "NICKNAME" )
 			d->nickName = i.text().trimmed();
 		else if ( tag == "PHOTO" ) {
-			d->photo = QCA::Base64().stringToArray(subTagText(i, "BINVAL").replace("\n","")).toByteArray();
+			d->photo = QCA::Base64().stringToArray(subTagText(i, "BINVAL").replace(QRegExp("[\r\n]+"),"")).toByteArray();
 			d->photoURI = subTagText(i, "EXTVAL");
 		}
 		else if ( tag == "BDAY" )
