Index: talk/base/openssladapter.cc
===================================================================
--- talk/base/openssladapter.cc	(revision 1123282)
+++ talk/base/openssladapter.cc	(revision 1123701)
@@ -648,7 +648,11 @@
     int extension_nid = OBJ_obj2nid(X509_EXTENSION_get_object(extension));
 
     if (extension_nid == NID_subject_alt_name) {
+#if OPENSSL_VERSION_NUMBER >= 0x1000000fL
+      const X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+#else
       X509V3_EXT_METHOD* meth = X509V3_EXT_get(extension);
+#endif
       if (!meth)
         break;
 
