Index: talk/base/physicalsocketserver.cc
===================================================================
--- talk/base/physicalsocketserver.cc	(Revision 1206432)
+++ talk/base/physicalsocketserver.cc	(Arbeitskopie)
@@ -230,6 +230,7 @@
     // This is not possible on OSX.
     return -1;
 #endif
+    return 0;
   }
 
   int Send(const void *pv, size_t cb) {
@@ -372,6 +373,7 @@
 #endif   // __linux
 
     // TODO: OSX support
+    return 0;
   }
 
   SocketServer* socketserver() { return ss_; }
